core-bundle/src/Routing/ResponseContext/HtmlHeadBag/HtmlHeadBag.php

```diff
@@ -89,11 +91,20 @@ public function getCanonicalUri(): string
         return $this->canonicalUri;
     }
 
-    public function getCanonicalUriForRequest(Request $request): string
+    public function getCanonicalUriForRequest(Request $request, InsertTagParser|null $insertTagParser = null): string
     {
         if ($this->canonicalUri) {
+            if (null !== $insertTagParser) {
+                $url = $insertTagParser->replaceInline($this->canonicalUri);
```

Comment:
@ausi can we somehow ensure that we do not get any encoded values back here with the new service? 🤔

Reply:
Not yet, but it should be a possible extension because the insert tags specify the type of the output now.
What would you suggest or expect here? Something like `->replaceInline($this->canonicalUri, OutputType::url)`?

Reply:
Exactly, that would make sense so I can specify the correct context, right?